src/common/hooks/useGetPlays.js

```diff
@@ -0,0 +1,42 @@
+
+import { gql, useQuery } from '@apollo/client';
+
+const GET_PLAYS_QUERY = gql`
+query Fetch_Plays {
+  plays {
+    blog
+    component
+    cover
+    created_at
+    description
+    featured
+    github
+    id
+    language
+    level {
+      name
+    }
+    name
+    path
+    play_tags {
+      tag {
+        name
+      }
+    }
+    updated_at
+    user {
+      id
+      displayName
+      avatarUrl
+    }
+    video
+  }
+}
+`;
+
+const useGetPlays = () => {
```

Comment:
## TODO 1:

- We need to enhance this hook to take the page number, order by(sorting), and the condition(where clause) to get us plays that are paginated, sorted, and satisfy a query.
- Should we pass the search and filter criteria as the where clause to this hook itself so that we do not pull all the plays on the client-side and then do the filtering. Instead, we do the filtering at the server-side itself. I strongly feel so. @koustov any suggestion here?
- In the long run, we want to show the first 20 plays and then scroll and get the rest. Should we do the pagination in this PR itself or later? I am fine to do it later as the play list page will undergo design changes.

Reply:
i have some thoughts on it. we can discuss once we connect.

Reply:
I am looking it from a time perspective, lets the first cut be very simple and mimic the current behavior and incrementally work on it for pagination/yield.